cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -424,7 +424,9 @@ private MetadataSearchResponse search(Set<MetadataScope> scopes, String namespac
         maxEndIndex = 0;
       } else {
         startIndex = offset;
-        maxEndIndex = offset + limit;
+        // account for overflow
+        long testEndIndex = (long) offset + (long) limit;
```

Comment:
You do, otherwise `offset + limit` will be calculated as an `int` before being upcast to `long`, which isn't what we want.